Mathlib/Analysis/SpecialFunctions/Integrals/PosLogEqCircleAverage.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stefan Kebekus
+-/
+import Mathlib.Analysis.SpecialFunctions.Integrals.Basic
+import Mathlib.Analysis.SpecialFunctions.Integrals.LogTrigonometric
+import Mathlib.MeasureTheory.Integral.CircleAverage
+
+/-!
+# Representation of `log⁺` as a Circle Average
+
+If `a` is any complex number of norm one, establish that the circle average
+`circleAverage (log ‖· - a‖) 0 1` vanishes.
+
+TODO: As soon as the mean value theorem for harmonic functions becomes available, extend this result
+to arbitrary complex numbers `a`, showing that the circle average equals the positive part of the
+logarithm, `circleAverage (log ‖· - a‖) 0 1 = = log⁺ ‖a‖`. This result, in turn, is a major
+ingredient in the proof of Jensen's formula in complex analysis.
+-/
+
+open Filter Interval intervalIntegral MeasureTheory Real
+
+variable {a : ℂ}
+
+/-!
+## Computing `circleAverage (log ‖· - a‖) 0 1` in case where `‖a‖ = 1`.
+-/
+
+/--
+If `a : ℂ` has norm one, then the circle average `circleAverage (log ‖· - a‖) 0 1` vanishes.
+-/
+@[simp]
+theorem circleAverage_log_norm_id_sub_const₁ (h : ‖a‖ = 1) :
+    circleAverage (log ‖· - a‖) 0 1 = 0 := by
+  simp only [circleAverage, mul_inv_rev, smul_eq_mul, mul_eq_zero, inv_eq_zero, OfNat.ofNat_ne_zero,
+    or_false]
+  right
+  obtain ⟨ζ, hζ⟩ := (by simp [Set.exists_range_iff.1, h] : ∃ ζ, a⁻¹ = circleMap 0 1 ζ)
+  calc ∫ x in 0..(2 * π), log ‖circleMap 0 1 x - a‖
+  _ = ∫ x in 0..(2 * π), log ‖(circleMap 0 1 ζ) * (circleMap 0 1 x - a)‖ := by
+    simp
+  _ = ∫ x in 0..(2 * π), log ‖circleMap 0 1 (ζ + x) - (circleMap 0 1 ζ) * a‖ := by
+    simp [mul_sub, circleMap, add_mul, Complex.exp_add]
+  _ = ∫ x in 0..(2 * π), log ‖circleMap 0 1 (ζ + x) - 1‖ := by
+    simp [← hζ, inv_mul_cancel₀ (by aesop : a ≠ 0)]
+  _ = ∫ x in 0..(2 * π), log ‖circleMap 0 1 x - 1‖ := by
+    have : Function.Periodic (log ‖circleMap 0 1 · - 1‖) (2 * π) :=
+      fun x ↦ by simp [periodic_circleMap 0 1 x]
+    have := this.intervalIntegral_add_eq (t := 0) (s := ζ)
+    simp_all [integral_comp_add_left (log ‖circleMap 0 1 · - 1‖)]
+  _ = ∫ x in 0..(2 * π), log (4 * sin (x / 2) ^ 2) / 2 := by
```

Comment:
Could you maybe extract the value of this integral in a lemma before this one, to separate clearly what happens in the complex plane and what is just an integral computation?

Reply:
Implemented as suggested.